mod-data-import-converter-storage-server/src/main/java/org/folio/rest/impl/DataImportProfilesImpl.java

```diff
@@ -678,9 +687,13 @@ public void deleteDataImportProfilesActionProfilesById(String id, String lang, M
     vertxContext.runOnContext(v -> {
       try {
         OkapiConnectionParams params = new OkapiConnectionParams(okapiHeaders, vertxContext.owner());
-        actionProfileService.markProfileAsDeleted(id, params.getTenantId())
-          .map(DeleteDataImportProfilesActionProfilesByIdResponse.respond204WithTextPlain(
-            format("Action Profile with id '%s' was successfully deleted", id)))
+        actionProfileService.isProfileAssociatedAsDetail(id, params.getTenantId())
+          .compose(isHasAssociations -> isHasAssociations
+            ? Future.succeededFuture(DeleteDataImportProfilesActionProfilesByIdResponse
+              .respond409WithTextPlain(format(DELETE_PROFILE_ERROR_MESSAGE, id)))
+            : actionProfileService.markProfileAsDeleted(id, tenantId)
+              .map(DeleteDataImportProfilesActionProfilesByIdResponse.respond204WithTextPlain(
+                format("Action Profile with id '%s' was successfully deleted", id))))
           .map(Response.class::cast)
```

Comment:
To avoid code duplication and poor formating All above refactor with these:
```
private Function<Boolean, Future<Response>> hasAssociationsResponseMapper(String id, ContentType profileType, OkapiConnectionParams params) {
    return hasAssociations -> {
      Future<Response> result;
      if (hasAssociations) {
        result = Future.succeededFuture(get409Response(profileType, format(DELETE_PROFILE_ERROR_MESSAGE, id)));
      } else {
        result = mappingProfileService.markProfileAsDeleted(id, params.getTenantId())
          .map(get204Response(profileType, format(DELETE_PROFILE_SUCCESS_MESSAGE, id)));
      }
      return result;
    };
  }

  private Response get204Response(ContentType profileType, String message) {
    switch (profileType) {
      case MAPPING_PROFILE:
        return DeleteDataImportProfilesMappingProfilesByIdResponse.respond204WithTextPlain(message);
      case ACTION_PROFILE:
        return DeleteDataImportProfilesActionProfilesByIdResponse.respond204WithTextPlain(message);
      case MATCH_PROFILE:
        return DeleteDataImportProfilesMatchProfilesByIdResponse.respond204WithTextPlain(message);
      default:
        throw new NotSupportedException("Cant map response for ...");
    }
  }

  private Response get409Response(ContentType profileType, String message) {
    switch (profileType) {
      case MAPPING_PROFILE:
        return DeleteDataImportProfilesMappingProfilesByIdResponse.respond409WithTextPlain(message);
      case ACTION_PROFILE:
        return DeleteDataImportProfilesActionProfilesByIdResponse.respond409WithTextPlain(message);
      case MATCH_PROFILE:
        return DeleteDataImportProfilesMatchProfilesByIdResponse.respond409WithTextPlain(message);
      default:
        throw new NotSupportedException("Cant map response for ...");
    }
  }

```